### PR TITLE
route: Accept CIDR notation when specifying network

### PR DESCRIPTION
**Pull request by brapru**
Now that the IPv4Address has the ability to generate valid IP addresses from CIDR notations, this provides a nicer interface to the user when specifying the network address.

For example:
`$ route add --net 8.8.8.0/24 --gateway 192.168.86.1 --interface ep0s8`

### CODE REVIEW

File: Userland/Utilities/route.cpp [hunk 1]
```diff
@@ -157,8 +157,18 @@ ErrorOr<int> serenity_main(Main::Arguments arguments)
...
-        if (!value_network_address.is_empty())
-            destination = AK::IPv4Address::from_string(value_network_address);
+        StringView address;
+        StringView cidr;
+        if (!value_network_address.is_empty()) {
+            // Check if a CIDR notation was provided and parse accordingly
+            if (auto position = value_network_address.find('/'); position.has_value()) {
+                address = value_network_address.substring_view(0, position.value());
+                cidr = value_network_address.substring_view(position.value() + 1);
+            } else
```
Comment (ADKaster) [inline Userland/Utilities/route.cpp:167]: style issue here, if the `if` block has braces, the `else` should as well

Comment (brapru) [reply Userland/Utilities/route.cpp]: Ah, yes. Thanks!

File: Userland/Utilities/route.cpp [hunk 2]
```diff
@@ -157,8 +157,18 @@ ErrorOr<int> serenity_main(Main::Arguments arguments)
...
 
-        if (!value_network_address.is_empty())
-            destination = AK::IPv4Address::from_string(value_network_address);
+        StringView address;
+        StringView cidr;
+        if (!value_network_address.is_empty()) {
+            // Check if a CIDR notation was provided and parse accordingly
+            if (auto position = value_network_address.find('/'); position.has_value()) {
+                address = value_network_address.substring_view(0, position.value());
+                cidr = value_network_address.substring_view(position.value() + 1);
```
Comment (linusg) [inline Userland/Utilities/route.cpp:166]: Does this still work for input like `1.2.3.4/`, where `/` is the last character?

File: Userland/Utilities/route.cpp [hunk 3]
```diff
@@ -171,7 +181,12 @@ ErrorOr<int> serenity_main(Main::Arguments arguments)
             return 1;
         }
 
-        auto genmask = AK::IPv4Address::from_string(value_netmask_address);
+        Optional<IPv4Address> genmask;
+        if (cidr.to_int().has_value())
+            genmask = AK::IPv4Address::netmask_from_cidr(cidr.to_int().value());
```
Comment (linusg) [inline Userland/Utilities/route.cpp:186]:
```suggestion
        if (auto cidr_int = cidr.to_int(); cidr_int.has_value())
            genmask = AK::IPv4Address::netmask_from_cidr(cidr_int.value());
```

Comment (brapru) [reply Userland/Utilities/route.cpp]: :+1:

File: Userland/Utilities/route.cpp (hunk 2 above)
Comment (brapru) [reply Userland/Utilities/route.cpp:166]: Yeah, it does. It won't pass the `cidr_int.has_value()` conditional check below, so an empty string will be passed to generate the genmask `IPv4Address`. That will also fail and then the program will exit on failure.

**Pull request merged**